ecommerce_integrations/unicommerce/inventory.py

```diff
@@ -86,3 +90,65 @@ def _update_inventory_sync_status(ecom_item_success_map: Dict[str, bool], timest
 	for ecom_item, status in ecom_item_success_map.items():
 		if status:
 			update_inventory_sync_status(ecom_item, timestamp)
+
+
+def shelf_bulk_update(warehouse, settings):
+	warehouse = warehouse["erpnext_warehouse"]
+	shelves = frappe.get_list("Shelf", {"warehouse": warehouse, "disable": 0}, ["shelf_name", "type"])
+	is_group_warehouse = cint(frappe.db.get_value("Warehouse", warehouse, "is_group"))
+	if is_group_warehouse:
+		erpnext_inventory = get_inventory_levels_of_group_warehouse(
+			warehouse=warehouse, integration=MODULE_NAME
+		)
+	else:
+		erpnext_inventory = get_inventory_levels(warehouses=(warehouse,), integration=MODULE_NAME)
+
+	if not erpnext_inventory:
+		return
+
+	# erpnext_inventory = erpnext_inventory[:MAX_INVENTORY_UPDATE_IN_REQUEST]
+	report = frappe.get_doc("Report", "Stock Ledger")
+	wh_to_facility_map = settings.get_erpnext_to_integration_wh_mapping()
+	facility_code = wh_to_facility_map[warehouse]
+	inventory_list = []
+	for shelf in shelves:
+		inventoryType = "GOOD_INVENTORY"
+		if shelf.type == "Unsellable":
+			inventoryType = "BAD_INVENTORY"
+		for item in erpnext_inventory:
+			custom_filter = {
+				"company": "Lifelong Online Retail Private Limited",
+				"from_date": datetime.today().strftime("%Y-%m-%d"),
+				"to_date": datetime.today().strftime("%Y-%m-%d"),
+				"warehouse": warehouse,
+				"shelf": shelf.shelf_name,
+				"item_code": item.item_code,
+			}
+			columns, data = report.get_data(limit=1, filters=custom_filter, as_dict=True)
+			if len(data) > 0:
+				data = data[0]
+				inventory_list.append(
+					{
+						"itemSKU": item.item_code,
+						"quantity": data["qty_after_transaction"],
+						"shelfCode": shelf,
```

Comment:
Here underlying assumption is that shelf name on unicommerce is same as what it is on ERPNext. It should instead be a mapping like warehouse->location.

Reply:
Removed